### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by Donmark2k**
### Blog app - Add forms
In this branch, I implemented the following
> Use preprocessed HTML file with embedded Ruby code.
> Use layouts and templates for shared content.

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,9 @@
...
\ No newline at end of file
+<h1>List of users </h1>
+<% @users.each do |user| %>
+    <%= render 'shared/user', user:user%>
+<%end%>
+<div class="button_holder">
+<button>
+<%= link_to 'Create new post', new_user_post_path(@current_user)%>
+</button>
+</div>
```
Comment (rotimiazeez) [inline app/views/users/index.html.erb]: When your app starts, it renders an error in your root path. Please look into it 👍🏽
Screenshot from 2023-05-13 10-37-46

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @Donmark2k  👍🏽,
## Good job so far!
### To Highlight 👇🏽
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @Donmark2k  👍🏽,
## Good job so far!
### To Highlight 👇🏽
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: app/views/users/index.html.erb (hunk 1 above)
Comment (Donmark2k) [reply app/views/users/index.html.erb]: https://www.loom.com/share/974abb95d2ed4518a832671cc062b20a

Comment (Donmark2k) [reply app/views/users/index.html.erb]: The application is starting from my own end

Comment (Donmark2k) [reply app/views/users/index.html.erb]: I am open to a zoom call or slack message, I am currently online now

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,10 +1,30 @@
...
+
+  def create
+    @post = @current_user.posts.new(post_params)
+    if @post.save
+      @user = User.find(params[:user_id])
+      redirect_to user_posts_path(@user)
+    else
+      render :new
+    end
+  end
```
Comment (elisha2kyakpo1) [inline app/controllers/posts_controller.rb]: - [ ] Kindly troubleshoot the post action and view to make it work, I tried to create a post but I failed, the app doesn't show any error. I have attached a screenshot for your reference.

posts

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @Donmark2k, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

👍 You did well to create a video to explain your app.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Donmark2k) [reply app/controllers/posts_controller.rb]: @elisha2kyakpo1 . if the first reviewer make changes, is the other changes not supposed to come as optional

Comment (elisha2kyakpo1) [reply app/controllers/posts_controller.rb]: @Donmark2k, thank you for reaching out. The first reviewer did not test your app, the issue was about when the app loads, he was getting an error. It was difficult for him to create a post/comment or like.

Comment (Donmark2k) [reply app/controllers/posts_controller.rb]: Thank you, I have rectified the changes

**Review by Reem-lab — APPROVED** [review]
# Hi @Donmark2k, 🤩🔥 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- [ ] Great job on the functionaliy for the forms 💯
- [ ] Professional README 🔥
- [ ] Followed Gitflow 🔥
- [ ] Great job working on the required changes 🔥🔥

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**